uniffi_bindgen/src/interface/types/mod.rs

```diff
@@ -146,10 +152,20 @@ impl From<&Type> for FFIType {
             | Type::Map(_)
             | Type::Timestamp
             | Type::Duration => FFIType::RustBuffer,
+            Type::External { .. } => FFIType::RustBuffer,
         }
     }
 }
 
+// The "kind" of external type.
+#[derive(Debug, Clone, Eq, PartialEq, Hash, Ord, PartialOrd)]
+pub enum ExternalTypeKind {
```

Comment:
Reading the code, it's not obvious to me what we want to do differently between the different kinds of external type here. I presume it's something like:

* For `UniFFI` types, we intend to use the generated scaffolding/trait impls from the external crate.
* The `Wrapping` types, we intend to generating our own scaffolding/trait impls.

Is there more to it than that?

It's also not obvious to me how much the two cases will have in common (which, I presume they will have much in common because otherwise we'd make them two different variants of `Type` enum..?).